src/rantanplan/rhymes.py

```diff
@@ -0,0 +1,190 @@
+        if relaxation:
+            ending = "".join(WEAK_VOWELS_RE.sub(r"\1", syll, count=1)
+                             for syll in stressed_ending)
+        else:
+            ending = "".join(stressed_ending)
+        ending = DIPHTHONG_Y_RE.sub(r"\1i\2", ending)
+        if assonance:
+            ending = CONSONANTS_RE.sub(r"", ending)
+        else:
+            # Consonance
+            ending = DIPHTHONG_H_RE.sub(r"\1\2", ending)
+            ending = INITIAL_CONSONANTS_RE.sub(r"", ending, count=1)
+        ending = strip_accents(ending)
+        if ending not in codes:
+            codes[ending] = len(codes)
+            unique.add(codes[ending])
+        else:
+            unique.discard(codes[ending])
+        code_numbers.append(codes[ending])
+    code2endings = {v: k for k, v in codes.items()}
+    # Adjust for free verses and assign letter codes
+    letters = {}
+    rhymes = []
+    endings = []
+    last_found = {}
+    for index, rhyme in enumerate(code_numbers):
+        if rhyme in unique:
+            rhyme_letter = -1  # free verse
+            endings.append("")
+        else:
+            if rhyme not in letters:
+                letters[rhyme] = len(letters)
+            rhyme_letter = letters[rhyme]
+            # Reassign free verses if an offset is set
+            if (rhyme in last_found
+                    and offset is not None
+                    and index - last_found[rhyme] > offset):
+                rhymes[last_found[rhyme]] = -1  # free verse
+                endings[last_found[rhyme]] = ''
+            last_found[rhyme] = index
+            endings.append(code2endings[rhyme])
+        rhymes.append(rhyme_letter)
+    # Reorder letters
+    rhyme_letters = []
+    letters = {}
+    stresses = []
+    for rhyme in rhymes:
+        if rhyme < 0:  # free verse
+            rhyme_letter = free_verse_symbol
+        else:
+            if rhyme not in letters:
+                letters[rhyme] = len(letters)
+            rhyme_letter = string.ascii_letters[letters[rhyme]]
+        rhyme_letters.append(rhyme_letter)
+    # Extract stress
+    stresses = []
+    for index, ending in enumerate(endings):
+        if not ending:
+            stresses.append(0)
+        ending_lower = ending.lower()
+        if ending_lower != ending:
+            positions = [pos - len(ending)
+                         for pos, char in enumerate(ending)
+                         if char.isupper()]
+            stresses.append(positions[0])  # only return first stress detected
+            endings[index] = ending_lower
+    return rhyme_letters, endings, stresses
+
+
+def search_structure_index(rhyme, syllables_count, structure_key):
+    for index, (key, _, structure, func) in enumerate(STRUCTURES):
+        if (key == structure_key
+                and re.compile(structure).match(rhyme)
+                and func(syllables_count)):
+            return index
+
+
+def get_analysis(scansion, offset=4):
```

Comment:
Same name as the api function to get the full analysis, don't know if it will cause trouble.

Reply:
Will change.